flow/layers/clip_path_layer.cc

```diff
@@ -22,6 +22,7 @@ ClipPathLayer::~ClipPathLayer() = default;
 void ClipPathLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
   SkRect previous_cull_rect = context->cull_rect;
   SkRect clip_path_bounds = clip_path_.getBounds();
+  context->mutators_stack.pushClipPath(clip_path_);
```

Comment:
nit: move these inside the condition (if the clip is empty we're not painting the subtree), it's going to have the same effect only that the intention is more explicit.

Same for the other clip layers.

Reply:
done